src/transformers/hf_argparser.py

```diff
@@ -269,7 +283,7 @@ def parse_args_into_dataclasses(
         look_for_args_file=True,
         args_filename=None,
         args_file_flag=None,
-    ) -> Tuple[DataClass, ...]:
+    ) -> Tuple[T, ...]:
```

Comment:
To me this looks the same as annotating the return type to `Tuple[Any, ...]`, since `T` isn't used anywhere else.

The usual use of `TypeVar` is to infer an output type from either an input type (if `T` is used as annotation for an argument, like `identity(x: T) -> T`) or from a concretized version of a generic class (like `__getitem__()` return `T` for a `list[T]`).

I have some ideas for suggestions that I can leave in a comment!

Reply:
Thanks so much for the comment. I think the most ideal workflow is something like below working, but not sure if it's possible 👀

```

class HfArgumentParser:
    def __init__(self, test: List[Type[T]]):
        self.test = test

    def parse_args_into_dataclasses(self) -> List[T]:
        return self.test
    
parser = HfArgumentParser([RewardConfig, Config2])
args, args2 = parser.parse_args_into_dataclasses()
```

Reply:
The first option I listed below should do this exactly! The trick is that `HfArgumentParser` needs to inherit from `Generic[T]`.